Use better defaults for pageserver Docker image

SomeoneToIgnore opened the pull request:
Follow-up of https://github.com/neondatabase/neon/pull/2272 and https://neondb.slack.com/archives/C039YKBRZB4/p1660811291662979?thread_ts=1660773828.605099&cid=C039YKBRZB4

`init` overhaul [removed](https://github.com/neondatabase/neon/pull/2272/files#diff-79738685a656fe6b25061bb14181442210b599f746faeaba408a2401de45038a) the entrypoint that improved pageserver's usability inside the Docker container: it was able to start without errors and pretend to work.
PR recreates similar situation, but without an entrypoint.

hlinnaka commented:
Hmm, do we use the docker image for anything? When we deploy the pageserver, I believe we just extract the binary from the image. (Which seems a bit silly, but that's a different topic). Perhaps we should stop pretending that the docker image is "runnable", if we never actually run it? Or is that exactly what this PR does? :-)

SomeoneToIgnore commented:
In a way, yes: before, I've removed the entrypoint and now replaced the command and now nothing by default is run in the storage image.

Yet, people might want to try and run the entire storage inside Docker/compose/k8s after all, so I've updated the image to hold some non-invalid defaults, so it's easy to start and check things in case needed, similar to how e2e tests are launched.

hlinnaka commented:
> In a way, yes: before, I've removed the entrypoint and now replaced the command and now nothing by default is run in the storage image.
> 
> Yet, people might want to try and run the entire storage inside Docker/compose/k8s after all, so I've updated the image to hold some non-invalid defaults, so it's easy to start and check things in case needed, similar to how e2e tests are launched.

Ok. Can you add a comment at the top of the Dockefile to explain that?
